xen/arch/arm/coproc/plat/coproc_xxx.c

```diff
@@ -121,96 +85,41 @@ static const struct mmio_handler_ops vcoproc_xxx_mmio_handler = {
 
 s_time_t vcoproc_xxx_ctx_switch_from(struct vcoproc_instance *curr)
 {
-    s_time_t wait_time;
-
-    if ( !curr )
-        return 0;
-
-    ASSERT(curr->state == VCOPROC_RUNNING ||
-           curr->state == VCOPROC_ASKED_TO_SLEEP);
-
-    wait_time = NOW() & 1 ? coproc_wait_time : 0; /* random for now */
-
-    if ( wait_time == 0 )
-    {
-        if (curr->state == VCOPROC_RUNNING)
-            curr->state = VCOPROC_WAITING;
-        else
-            curr->state = VCOPROC_SLEEPING;
-    }
-
-    return wait_time;
+    /* random for now */
+    return NOW() & 1 ? coproc_wait_time : 0;
 }
 
 static int vcoproc_xxx_ctx_switch_to(struct vcoproc_instance *next)
 {
-    if ( !next )
-        return 0;
-
-    ASSERT(next->state == VCOPROC_WAITING);
-
-    next->state = VCOPROC_RUNNING;
-
+    /* nothing to do */
     return 0;
 }
 
-static struct vcoproc_instance *vcoproc_xxx_vcoproc_init(struct domain *d,
-                                                         struct coproc_device *coproc_xxx)
+static int vcoproc_xxx_vcoproc_init(struct domain *d,
+                                    struct coproc_device *coproc,
+                                    struct vcoproc_instance *vcoproc)
 {
-    struct vcoproc_instance *vcoproc_xxx;
     int i;
 
-    vcoproc_xxx = xzalloc(struct vcoproc_instance);
-    if ( !vcoproc_xxx )
+    for ( i = 0; i < coproc->num_mmios; i++ )
     {
-        dev_err(coproc_xxx->dev, "failed to allocate vcoproc_instance\n");
-        return ERR_PTR(-ENOMEM);
-    }
-
-    vcoproc_xxx->coproc = coproc_xxx;
-    vcoproc_xxx->domain = d;
-    vcoproc_xxx->state = VCOPROC_UNKNOWN;
-    spin_lock_init(&vcoproc_xxx->lock);
-
-    for ( i = 0; i < coproc_xxx->num_mmios; i++ )
-    {
-        struct mmio *mmio = &coproc_xxx->mmios[i];
+        struct mmio *mmio = &coproc->mmios[i];
         register_mmio_handler(d, &vcoproc_xxx_mmio_handler,
                               mmio->addr, mmio->size, mmio);
     }
 
-    spin_lock(&coproc_xxx->vcoprocs_lock);
-    list_add(&vcoproc_xxx->vcoproc_elem, &coproc_xxx->vcoprocs);
-    spin_unlock(&coproc_xxx->vcoprocs_lock);
-
-    return vcoproc_xxx;
+    return 0;
 }
 
 static void vcoproc_xxx_vcoproc_deinit(struct domain *d,
                                        struct vcoproc_instance *vcoproc_xxx)
 {
-    struct coproc_device *coproc_xxx;
-
-    if ( !vcoproc_xxx )
-        return;
-
-    coproc_xxx = vcoproc_xxx->coproc;
-    spin_lock(&coproc_xxx->vcoprocs_lock);
-    list_del(&vcoproc_xxx->vcoproc_elem);
-    spin_unlock(&coproc_xxx->vcoprocs_lock);
-    xfree(vcoproc_xxx);
-}
-
-static bool_t coproc_xxx_vcoproc_is_created(struct domain *d,
-                                            struct coproc_device *coproc_xxx)
-{
-    return coproc_xxx_get_vcoproc(d, coproc_xxx) ? true : false;
+    /* nothing to do */
```

Comment:
if it seems to be a reference code, please put a comment inside, what should be done here